Tests/LibC/TestPwd.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2022, Andreas Kling <kling@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibTest/TestCase.h>
+
+#include <AK/Vector.h>
+#include <pwd.h>
+
+struct PasswdEntry {
+    String name;
+    uid_t uid {};
+};
+
+static Vector<PasswdEntry> get_all_passwd_entries()
+{
+    Vector<PasswdEntry> entries;
+    setpwent();
+    while (auto* pwd = getpwent()) {
```

Comment:
Was watching your [YouTube video](https://www.youtube.com/watch?v=fk0EMHevbPs) and was wondering if this call to `getpwent()` makes sense as a call to `getpwent_r()` instead? Full disclosure that I'm not sure what the consequence of not using the re-entrant api here would be, but I was curious and figured I'd leave a comment :)

Reply:
Update: ah, never mind. It looks like the implementation of [`getpwent`](https://github.com/SerenityOS/serenity/blob/8363225320f9cb3fc79e6e9916bb6d441a04552c/Userland/Libraries/LibC/pwd.cpp#L125) uses [`getpwent_r`](https://github.com/SerenityOS/serenity/blob/8363225320f9cb3fc79e6e9916bb6d441a04552c/Userland/Libraries/LibC/pwd.cpp#L130-L167)!